packages/server/graphql/subscribeGraphQL.ts

```diff
@@ -76,14 +78,20 @@ const subscribeGraphQL = async (req: SubscribeRequest) => {
   // TODO PR definitelytyped
   for await (const payload of responseStream) {
     const {data} = payload
-    if (data?.notificationSubscription?.__typename === 'AuthTokenPayload') {
+    const notificationType = data?.notificationSubscription?.__typename
+    if (notificationType === 'AuthTokenPayload') {
       const jwt = (data.notificationSubscription as IAuthTokenPayload).id
       connectionContext.authToken = new AuthToken(decode(jwt) as any)
       // if auth changed, then we can't trust any of the subscriptions, so dump em all and resub for the client
       // delay it to guarantee that no matter when this is published, it is the last message on the mutation
       setTimeout(() => {
         relayUnsubscribeAll(connectionContext, {isResub: true})
       }, 1000)
+    } else if (notificationType === 'InvalidateSessionsPayload') {
+      handleDisconnect(connectionContext, {
```

Comment:
if this socket got their session invalidated, disconnect them immediately before they can make an unvalidated request